src/WebJobs.Script.WebHost/Security/ISecretsRepository.cs

```diff
@@ -17,6 +17,10 @@ public interface ISecretsRepository
 
         Task WriteAsync(ScriptSecretsType type, string functionName, string secretsContent);
 
+        Task WriteSnapshotAsync(ScriptSecretsType type, string functionName, string secretsContent);
+
         Task PurgeOldSecretsAsync(IList<string> currentFunctions, TraceWriter traceWriter, ILogger logger);
+
+        Task<string[]> ListSecretsByPrefixAsync(string prefix);
```

Comment:
The latest iteration looks great. The only thing I'd recommend changing is to remove this method from the interface and replace it with a method to retrieve snapshots (e.g. `GetSecretSnapshots`, providing the type and function name). This remove the implicit naming contract that currently exists between the manager and repository implementations. It also remove the assumption that snapshots are handled with name changes, which will not always be the case (in the case of storage, for example, we could be taking actual snapshots of the blob).